content-services/content-sources-backend
# Fixes 2282: reduce number of introspect tasks

## Summary

This restructures the introspection code to move all decisions about whether a repo should be introspected into the main query to lookup repos for introspection.  the result is that we do not spawn tasks for repos that won't actually be introspected.  It should reduce the tasks we spawn by about 2/3rds.  

Some considerations:
* Now in pkg/external_repos  any introspect function will introspect without question
* The notifications are being sent one at a time for introspection, but they were before as well.  We likely want to revisit how those are generated.

## Testing steps

1.  Create one or more repos for introspection (but turn off snapshotting for simplicity)
2. run ```go run cmd/external-repos/main.go nightly-jobs```
3. notice that no tasks get run
4.  wait ~24 hours or adjust config.IntrospectTimeInterval in value_constraints.go to `time.Minute * 1`  and wait ~ 1 minute
5. run ```go run cmd/external-repos/main.go nightly-jobs```
6. notice taht tasks now get spawned 


## Checklist
